rclcpp/include/rclcpp/parameter.hpp

```diff
@@ -93,6 +95,30 @@ class ParameterVariant
 
   inline ParameterType get_type() const {return static_cast<ParameterType>(value_.type); }
 
+  inline std::string get_type_name() const
+  {
+    switch (get_type()) {
+      case rclcpp::parameter::ParameterType::PARAMETER_BOOL:
+        return "bool";
+        break;
+      case rclcpp::parameter::ParameterType::PARAMETER_INTEGER:
+        return "integer";
+        break;
+      case rclcpp::parameter::ParameterType::PARAMETER_DOUBLE:
+        return "double";
+        break;
+      case rclcpp::parameter::ParameterType::PARAMETER_STRING:
+        return "string";
+        break;
+      case rclcpp::parameter::ParameterType::PARAMETER_BYTES:
+        return "bytes";
+        break;
+      case rclcpp::parameter::ParameterType::PARAMETER_NOT_SET:
+      default:
```

Comment:
The default case and the not-set case should be separate to catch the problem if new types are added in the future.